### ISSUE TITLE
Registration for translated event not shown on page "my events"

### ISSUE THREAD

**Issue by christophlehmann** [derhansen/sf_event_mgt#821]
### Extension version

5.2.0

### Short problem description

I registered to a translated event in the translated language (1 = EN) and i don't see this registration on the page "my events".

### Steps to reproduce
1. Register to a translated event in non default language 
2. Visit page "my events" (The page containing my user registrations)

### Expected behaviour

I see my registration.

### Actual behaviour

I don't see my registration in the default language and in the translation

The field sys_language_uid of the registration record contains 1 (= EN)

**Comment by christophlehmann** [comment]
I currently see two possibilities to see my registration

i) Set sys_language_uid = 0 in registration

ii) Set `->setRespectSysLanguage(false);` (or remove it) in `\DERHANSEN\SfEventMgt\Domain\Repository\RegistrationRepository::findRegistrationsByUserRegistrationDemand()`

**Comment by derhansen** [comment]
Thanks for you report. I really hoped that 7a0416644f4e19aed3ac7a1cf18dcaf130a562ec had fixed all problems with the language of registrations/emails/field translations, but obviously not.

Both your suggested changes result in some unexpected behavior.

1) Setting `sys_language_uid = 0` to the registration will actually result in the situation, that you can not confirm the registration any more. If the recommended route enhancers are used, the parameter `reguid` can not be resolved any more, since the registration can not be fetched for the current language if frontend is in non-default language. Route enhancers may need to be adjusted, so the `reguid` parameter is just treated as a static value (also resulting a cHash in URLs).

2) When you set `->setRespectSysLanguage(false)` you will notice, that the event title in the "My events" plugin will be in the *default* language when the page language actually is the non default language.

So maybe option 1 is the way to go, but I will need to setup some more acceptance tests in order to make sure that nothing breaks. Especially for extbase actions, where `$registration` is used as action parameter (generally all actions in `PaymentController`) this change may be breaking.

Also I'm pretty unsure if this fix may introduce new problems in v 4.x of the extension when used in TYPO3 8.7 or 9.5.

So I'm afraid this issue will take some time to resolve (maybe during christmas holidays)

**Comment by christophlehmann** [comment]
Hello Torben,

I'm on TYPO3 v10.4

When i set `->setRespectSysLanguage(false)` i see the translated event title. For untranslated events i see the german title (default language), which is also fine, because that way i know which events i'm registered on independently of the language. This independence maybe unexpected, yes.

**Comment by derhansen** [comment]
I'm sorry, but I can not reproduce this behavior. I have an event (default language EN) and a translation for german (DE). I register to the event both in default and non default language.

Now I open the page with the user registration in a non default language (DE) and I only see one the event registration, where the event title is correctly translated

![image](https://user-images.githubusercontent.com/2629896/101533648-62c6b780-3996-11eb-8ad0-9c8c1c403ff9.png)

Now I change `->setRespectSysLanguage(false)` in `findRegistrationsByUserRegistrationDemand` and I see both registrations, but the event title changed to the default language

![image](https://user-images.githubusercontent.com/2629896/101533773-8ab61b00-3996-11eb-89b7-e6562ded47a0.png)

**Comment by derhansen** [comment]
Now I know why you have different results than me... I debugged this back at the beginning of the year and found out, that extbase does the language overlay depending on language of the *first* record in the query result. So if the first registration in the user registrations view has the *default language*, all following registrations will be overlaid in that language too.

**Comment by christophlehmann** [comment]
I tried some things...

Case 1) Use `->setRespectSysLanguage(false)`

* Event 1 (L=0) not translated
* Have registration to Event 1
* Event 2 (L=0 + 1)
* Have registration to Event 2 (L=1)

Visit Page (L=1) I see Event 1 Title in L=0; I see Event 2 Title in L=1
Visit Page (L=0) I see Event 1 Title in L=0; I see Event 2 Title in L=1 (not good)

=> I hide Registration to Event 1

Visit Page (L=1) I see Event 2 Title in L=1 
Visit Page (L=0) I see Event 2 Title in L=1 (not good)

Conclusion: The event title depends on the registrations language.

---

Case 2) Do not use language fields in registration

```
/*
        'languageField' => 'sys_language_uid',
        'transOrigPointerField' => 'l10n_parent',
        'transOrigDiffSourceField' => 'l10n_diffsource',
*/
```

I have

* Event 1 (L=0) not translated
* Have registration to Event 1
* Event 2 (L=0 + 1)
* Have registration to Event 2

Visit Page (L=1) I see Event 1 Title in L=0; I see Event 2 Title in L=1 (Fine so far - the best what i can see)
Visit Page (L=0) I see Event 1 Title in L=0; I see Event 2 Title in L=0 (better!)

=> I hide Registration to Event 1 (which makes no difference): 

Visit Page (L=1) I see Event 2 Title in L=1 (good)
Visit Page (L=0) I see Event 2 Title in L=0 (good)

We discussed the case already in #688

---

Case 2a)

I have Event 3 in L=1 (not translated)

Visit Page (L=0) I don't see Event 3 title (= Link to detail page)
Visit Page (L=1) I see  Event 3 title

I would neglect that case atm.

Case 3) 

* Use Language fields in registrations
* Use `->setRespectSysLanguage(true)`
* All my registrations have L=-1

It fully behaves like in case 2 which is good so far.

In the [commit](https://github.com/derhansen/sf_event_mgt/commit/956c6654c113e620cd6ced9eb7f29bdce31d80d7#diff-beb7f50797fbcca55ce92a337673cd1d40c2156c17ddb5ff0afffafa133cc31eR332) you mentioned that L=-1 makes problems with notifications. Maybe we can go with it (L=-1) and fix the notification problem?

**Comment by christophlehmann** [comment]
Side topic: With https://github.com/christophlehmann/sf_event_mgt/commit/5cf36845390488e05069cf9af41023de9696e00f the registration links may lead to an error, if you cancel in DE and the registration has sys_language_uid=EN. As a workaround i use the registrations language in the cancellation link.

**Comment by derhansen** [comment]
Closed with 1a66a25
